### PR TITLE
Ensure buff size constraints in chip logging module name fetch

### LINKED ISSUE #4730: GetModuleName can write out of bounds on its buffer
#### Problem
In ` src/lib/support/logging/CHIPLogging.cpp` we have, in `GetModuleName`:
```
     buf[chip::Logging::kMaxModuleNameLen] = 0;
```
but nothing ensures that `bufSize` is larger than `chip::Logging::kMaxModuleNameLen`.

 #### Proposed Solution
Probably just remove this line, since the immediately preceding line is `snprintf(buf, ....)` and that will always null-terminate (unlike the pre-#4627 code that did a memcpy).

That said, that `snprintf` call will also do the wrong thing if the incoming buffer size is _larger_ than `kMaxModuleNameLen + 1`: it will get part of the next module name.  We should be clamping `bufSize` to `kMaxModuleNameLen + 1`, I suspect.

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
Potential buffer overflow in CHIPLogging.cpp

#### Change overview
Fixes it.

Fixes #4730

#### Testing
Compiled and ran chip-tool, saw logs looking good.

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]
Thank you!

File: src/lib/support/logging/CHIPLogging.cpp [hunk 1]
```diff
@@ -102,10 +102,30 @@ static const char ModuleNames[] = "-\0\0" // None
 
 void GetModuleName(char * buf, uint8_t bufSize, uint8_t module)
 {
-    const char * moduleNamePtr = ModuleNames + ((module < ModuleNamesCount) ? module * chip::Logging::kMaxModuleNameLen : 0);
+    if (bufSize == 0)
+    {
+        return; // no space for anything
+    }
```
Comment (msandstedt) [inline src/lib/support/logging/CHIPLogging.cpp:108]: We can't just return here.  The caller is likely expecting a null-terminated string.  If we can't do that, we need to propagate an error to the caller.

Why don't we define and pass a CapacityBoundBuffer<length> type or similar so this code can know at compile time the size of the buffer it's going to write into?

Comment (andy31415) [reply src/lib/support/logging/CHIPLogging.cpp:108]: Should I die insead? I am being told that there is no space in the buffer, not even for null terminator.

**Review by msandstedt — APPROVED** [review]

File: src/lib/support/logging/CHIPLogging.cpp (hunk 1 above)
Comment (msandstedt) [reply src/lib/support/logging/CHIPLogging.cpp:108]: Also, doing what snprintf does for this case seems like a reasonable approach given the external interface of this function.

Comment (andy31415) [reply src/lib/support/logging/CHIPLogging.cpp]: Updated to enforce at compile time.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by andy31415** [comment]
size increases are not reasonable. Trying to re-trigger with master merge.

**Pull request merged**